root/healthcheck.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/with-contenv bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# universal check that the process is running
+pgrep -af nginx > /dev/null
+
+# individualized check(s) specific to the app
+nginx -tq -c /config/nginx/nginx.conf > /dev/null
```

Comment:
From a functionality standpoint , testing the config should be done in the init of the container and message the user and maybe block us from running Nginx, the test should be something functional for this specific container.

Reply:
I can agree with that. Like you said in your other comment I'm also not sure what would be a good test for this specific container. I mentioned testing the using the URL environment variable which I think would look something like this: `curl -f "https://${URL}/" >/dev/null`. I'm in agreement that input from the other devs is needed on this though.

Reply:
Why do a full curl? , wouldn't a 200 response code lookup be sufficient and leaner :
```
curl -ILs -w "%{http_code}" -o /dev/null google.com
```

Reply:
Indeed, that should work, good idea!